rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -689,1112 +552,1189 @@ fn induct_loopback_stream_with_memory_limit_impl(
+    expected = "Invalid signal indices in stream to subnet 5h3gz-qaxaa-aaaaa-aaaap-yai: signals_end 153, signals [RejectSignal { reason: CanisterMigrating, index: 138 }, RejectSignal { reason: CanisterNotFound, index: 139 }, RejectSignal { reason: CanisterStopped, index: 145 }, RejectSignal { reason: CanisterStopping, index: 142 }]"
 )]
 fn garbage_collect_signals_in_wrong_order() {
-    with_test_replica_logger(|log| {
-        let mut stream = generate_outgoing_stream(StreamConfig {
-            messages_begin: 31,
-            message_count: 3,
+    use RejectReason::*;
+    with_test_setup(
+        // An outgoing stream with reject signals out of order.
+        btreemap![REMOTE_SUBNET => StreamConfig {
             signals_end: 153,
-            // Reject signals not in order.
-            reject_signals: Some(vec![
-                RejectSignal::new(RejectReason::CanisterMigrating, 138.into()),
-                RejectSignal::new(RejectReason::CanisterMigrating, 139.into()),
-                RejectSignal::new(RejectReason::CanisterMigrating, 145.into()),
-                RejectSignal::new(RejectReason::CanisterMigrating, 142.into()),
-            ]),
-        });
-
-        let slice = generate_stream_slice(StreamSliceConfig {
-            header_begin: 142,
-            header_end: None,
+            reject_signals: vec![
+                RejectSignal::new(CanisterMigrating, 138.into()),
+                RejectSignal::new(CanisterNotFound, 139.into()),
+                RejectSignal::new(CanisterStopped, 145.into()),
+                RejectSignal::new(CanisterStopping, 142.into()),
+            ],
+            ..StreamConfig::default()
+        }],
+        // An incoming stream slice with `header_begin` such that some reject signals
+        // are garbage collected.
+        btreemap![REMOTE_SUBNET => StreamSliceConfig {
+            header_begin: Some(142),
             messages_begin: 153,
-            message_count: 10,
-            signals_end: 33,
-            reject_signals: None,
-            flags: Default::default(),
-        });
-
-        let mut stats = Default::default();
-        let (stream_handler, _, _) = new_fixture(&log);
-        stream_handler.garbage_collect_signals(
-            &mut StreamHandle::new(&mut stream, &mut stats),
-            REMOTE_SUBNET,
-            &slice,
-        );
-    });
+            ..StreamSliceConfig::default()
+        }],
+        |stream_handler, mut state, slices, _| {
+            let mut streams = state.take_streams();
+
+            stream_handler.garbage_collect_signals(
+                &mut streams.get_mut(&REMOTE_SUBNET).unwrap(),
+                REMOTE_SUBNET,
+                slices.get(&REMOTE_SUBNET).unwrap(),
+            );
+        },
+    );
 }
 
+/// Tests garbage collecting signals with a `StreamSlice` whose `message_begin` does not correspond
```

Comment:
Nit:
```suggestion
/// Tests that garbage collecting signals with a `StreamSlice` whose `message_begin` does not correspond
```

Same everywhere applicable.

Reply:
Done.